### PR TITLE
ESQL: Remove custom serializating infrastructure

### PR DESCRIPTION
**Pull request by nik9000**
Removes all of the remaining custom serialization infrastructure from ESQL, replacing it with `NamedWriteable`.

### CODE REVIEW

**Review by ChrisHegarty — APPROVED** [review]
Lovely! LGTM

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/ReferenceAttributeTests.java [hunk 1]
```diff
@@ -14,18 +14,20 @@
 import org.elasticsearch.xpack.esql.core.type.DataType;
 
 public class ReferenceAttributeTests extends AbstractAttributeTestCase<ReferenceAttribute> {
-    public static ReferenceAttribute randomReferenceAttribute() {
+    public static ReferenceAttribute randomReferenceAttribute(boolean onlyRepresentable) {
         Source source = Source.EMPTY;
         String name = randomAlphaOfLength(5);
-        DataType type = randomFrom(DataType.types());
+        DataType type = onlyRepresentable
```
Comment (astefan) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/ReferenceAttributeTests.java:20]: Why this change?

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/ReferenceAttributeTests.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/ReferenceAttributeTests.java:20]: I needed it for one of the serialization tests - they needed to only receive representable types from a `ReferenceAttribute`. I believe they needed to make a value of that type.

**Review by luigidellaquila — APPROVED** [review]
LGTM ❤️
